src/rshim.c

```diff
@@ -975,6 +979,33 @@ int rshim_fifo_sync(rshim_backend_t *bd)
       return rc;
   }
 
+  /* Drain the Rx FIFO until no more data in one second. */
+  if (drop_rx) {
+    avail = 0;
+
+    time(&t0);
+
+    do {
+      reg = 0;
+      rc = bd->read_rshim(bd, RSHIM_CHANNEL, bd->regs->tm_tth_sts, &reg, RSHIM_REG_SIZE_8B);
+      if (rc < 0 || RSHIM_BAD_CTRL_REG(reg))
+        break;
+
+      avail = reg & RSH_TM_TILE_TO_HOST_STS__COUNT_MASK;
+      if (avail == 0) {
+        time(&t1);
+        if (difftime(t1, t0) > 1)
+          break;
+        continue;
+      }
+
+      while (avail--)
+        bd->read_rshim(bd, RSHIM_CHANNEL, bd->regs->tm_tth_data, &reg, RSHIM_REG_SIZE_8B);
+
+      time(&t0);
```

Comment:
If there is no available data in the Rx FIFO, and the time difference (t1 - t0) is less than 1 second, the starting time t0 is refreshed. Does this mean: when avail = 0, the loop only exit if the time duration from the begining of the do-while loop to the time(&t1) is greater than 1 second?

Reply:
Yes, the worst case is up to 2 second which is ok.  Using ">=1" condition is a little risky since the gap might be too small (like the actual t0=0.9, t1=1.1, though time() function returns seconds, unless we use a different function)